src/main/java/programminglife/model/Graph.java

```diff
@@ -52,12 +52,13 @@ private void parseSegment(Scanner sc) {
         }
     }
 
-    void parseLink(Scanner sc) {
-        int sourceId = Integer.parseInt(sc.next());
-        sc.next();
-        int destinationId = Integer.parseInt(sc.next());
-        sc.next();
-        String overlap = sc.next();
+    void parseLink(String propertyString) {
+        String[] properties = propertyString.split("\\s");
+        // properties[0] is 'L'
```

Comment:
Instead if simply assuming, either add an assert (can we get Travis to run with assertions enabled?) or an explicit if
Add this as a precondition for the method in the JavaDoc either way.